### DISCUSSION TITLE
Forecasting new tabular data with dimension/shape differing from training shape

### DISCUSSION THREAD

**Discussion by msemenoff** [timeseriesAI/tsai#478] [Q&A] [answered]
First, thank you for this wonderful package.
I'm new to time series and have been working through the tsai regression tutorial with my own data.
My training data set shape is originally (2880, 18)
By following the data preparation tutorial and using the df2xy module, I transformed X and y to shapes (96, 30, 15) and (96, 30), respectively.
I was able to successfully train a RNNPlus model.
Now I want to use the trained model to predict 'targets' on new data. 
The new data shape is (1440, 17) (half the rows of the training set and, of course, no target).
This is where I'm stuck. I tried various combinations including a (96, 15, 15) shape using df2xy. My goal is to get predictions (1440, 1)
Thanks for any help.

**Comment by oguiza** [comment, marked as answer]
Hi @msemenoff,
Whenever you prepare new samples to generate predictions on, they need to have the same shape in axes 1 & 2. They can only differ on axis 0 (number of samples). Based on your comment above your new samples should have the shape (#samples, 30, 15). You can get to this by using the same df2Xy function but without the target col argument.

**Reply by msemenoff** [reply]
Hi oguiza,
Following your suggestion, I renamed and reordered my index and feature columns. Then, everything ran fine. Thank you for your reply.
